lib/parser/lexer.rl

```diff
@@ -833,6 +834,20 @@ class Parser::Lexer
     KEYWORDS_BEGIN[keyword] = KEYWORDS[keyword] = :"k#{keyword.upcase}"
   end
 
+  EMIT_INTEGER = lambda { |lexer, chars| lexer.emit(:tINTEGER,   chars) }
+  EMIT_RATIONAL = lambda { |lexer, chars| lexer.emit(:tRATIONAL,  Rational(chars)) }
+  EMIT_IMAGINARY = lambda { |lexer, chars| lexer.emit(:tIMAGINARY, Complex(0, chars)) }
+  EMIT_IMAGINARY_RATIONAL = lambda { |lexer, chars| lexer.emit(:tIMAGINARY, Complex(0, Rational(chars))) }
+  EMIT_INTEGER_RE = lambda { |lexer, chars| lexer.emit(:tINTEGER,   chars, @ts, @te - 2); p -= 2 }
```

Comment:
`p` is a local variable from `Lexer#advance`

Reply:
Oh, right.

Reply:
Fixed.